### PR TITLE
Add batch implementations for CDM (Synapse Link)

### PR DESCRIPTION
**Pull request by george-zubrienko**

### CODE REVIEW

File: framework/arcane-framework/src/main/scala/services/consumers/SynapseLink.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  override def reduceExpr: String =
+    // important to note that append-only nature of the source must be taken into account
+    // thus, we need identify which of the latest versions were deleted after we have found the latest versions for each `Id` - since for backfill we must exclude deletions
+    s"""SELECT * FROM (
+       | SELECT * FROM $name ORDER BY ROW_NUMBER() OVER (PARTITION BY Id ORDER BY versionnumber DESC) FETCH FIRST 1 ROWS WITH TIES
+       |) WHERE IsDelete = false""".stripMargin
+
+  override val batchQuery: OverwriteQuery = SynapseLinkBackfillQuery(targetName, reduceExpr)
+
+  def archiveExpr: String = s"INSERT OVERWRITE ${targetName}_stream_archive $reduceExpr"
```
Comment (s-vitaliy) [inline framework/arcane-framework/src/main/scala/services/consumers/SynapseLink.scala:50]: The archive table name should be a parameter

File: framework/arcane-framework/src/main/scala/services/consumers/SynapseLink.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  override def reduceExpr: String =
+    // for merge query, we must carry over deletions so they can be applied in a MERGE statement by MatchedAppendOnlyDelete
+    s"""SELECT * FROM (
+       | SELECT * FROM $name ORDER BY ROW_NUMBER() OVER (PARTITION BY Id ORDER BY versionnumber DESC) FETCH FIRST 1 ROWS WITH TIES
+       |)""".stripMargin
+
+  override val batchQuery: MergeQuery =
+    SynapseLinkMergeQuery(targetName = targetName, sourceQuery = reduceExpr, partitionValues = partitionValues, mergeKey = mergeKey, columns = schema.map(f => f.name))
+
+  def archiveExpr: String = s"INSERT INTO ${targetName}_stream_archive $reduceExpr"
```
Comment (s-vitaliy) [inline framework/arcane-framework/src/main/scala/services/consumers/SynapseLink.scala:71]: duplicate?

**Review by s-vitaliy — APPROVED** [review]

File: framework/arcane-framework/src/main/scala/services/consumers/SynapseLink.scala (hunk 1 above)
Comment (george-zubrienko) [reply framework/arcane-framework/src/main/scala/services/consumers/SynapseLink.scala:50]: Will send a PR for this

File: framework/arcane-framework/src/main/scala/services/consumers/SynapseLink.scala (hunk 2 above)
Comment (george-zubrienko) [reply framework/arcane-framework/src/main/scala/services/consumers/SynapseLink.scala:71]: No it's archive for a different batch type

**Pull request merged**